src/track/cueinfo.cpp:159:30: error: ambiguous overload for ‘operator<<’ (operand types are ‘QDebug’ and ‘std::optional<double>’) with qt-6.7

### Bug Description

This happens with qtbase git 6.7 branch (dev-qt/qtbase-6.7.9999) on Gentoo
```
[532/933] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DMIXXX_BUILD_RELEASE -DMIXXX_USE_QML -DMIXXX_USE_QOPENGL -DPA_USE_ALSA -DPROTOBUF_USE_DLLS -DQT_CONCURRENT_LIB -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_LABSQMLMODELS_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_PRINTSUPPORT_LIB -DQT_QMLBUILTINS_LIB -DQT_QMLINTEGRATION_LIB -DQT_QMLMODELS_LIB -DQT_QMLWORKERSCRIPT_LIB -DQT_QML_LIB -DQT_QUICKCONTROLS2IMPL_LIB -DQT_QUICKCONTROLS2_LIB -DQT_QUICKLAYOUTS_LIB -DQT_QUICKSHAPES_LIB -DQT_QUICKTEMPLATES2_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_SHADERTOOLS_LIB -DQT_SQL_LIB -DQT_SVGWIDGETS_LIB -DQT_SVG_LIB -DQT_TABLET_SUPPORT -DQT_TESTCASE_BUILDDIR=\"/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999_build\" -DQT_TESTCASE_SOURCEDIR=\"/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999\" -DQT_TESTLIB_LIB -DQT_USE_QSTRINGBUILDER -DQT_WIDGETS_LIB -DQT_XML_LIB -DSFC_SUPPORTS_SET_COMPRESSION_LEVEL -D__BATTERY__ -D__BULK__ -D__FAAD__ -D__FFMPEG__ -D__HID__ -D__KEYFINDER__ -D__LILV__ -D__LINUX__ -D__MAD__ -D__MODPLUG__ -D__MP4V2__ -D__OPUS__ -D__PORTMIDI__ -D__QTKEYCHAIN__ -D__RUBBERBAND__ -D__SNDFILE__ -D__SQLITE3__ -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D__UNIX__ -D__VINYLCONTROL__ -D__WV__ -Dx86_64 -I/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999_build/mixxx-lib_autogen/include -I/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src -I/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999_build/src -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/fidlib -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/portaudio -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/rigtorp/SPSCQueue/include -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/replaygain -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/reverb -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/kaitai -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/mp3guessenc-0.27.4 -isystem /usr/include/qt6/QtConcurrent -isystem /usr/include/qt6 -isystem /usr/include/qt6/QtCore -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtOpenGL -isystem /usr/include/qt6/QtPrintSupport -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtQml -isystem /usr/include/qt6/QtQmlIntegration -isystem /usr/include/qt6/QtQmlBuiltins -isystem /usr/include/qt6/QtSql -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtTest -isystem /usr/include/qt6/QtXml -isystem /usr/include/qt6/QtShaderTools -isystem /usr/include/qt6/QtSvgWidgets -isystem /usr/include/qt6/QtCore5Compat -isystem /usr/include/qt6/QtQuick -isystem /usr/include/qt6/QtQmlModels -isystem /usr/include/qt6/QtLabsQmlModels -isystem /usr/include/qt6/QtQmlModels/6.7.1 -isystem /usr/include/qt6/QtQmlModels/6.7.1/QtQmlModels -isystem /usr/include/qt6/QtCore/6.7.1 -isystem /usr/include/qt6/QtCore/6.7.1/QtCore -isystem /usr/include/qt6/QtQml/6.7.1 -isystem /usr/include/qt6/QtQml/6.7.1/QtQml -isystem /usr/include/qt6/QtQmlBuiltins/6.7.1 -isystem /usr/include/qt6/QtQmlBuiltins/6.7.1/QtQmlBuiltins -isystem /usr/include/qt6/QtQuickControls2 -isystem /usr/include/qt6/QtQuickControls2Impl -isystem /usr/include/qt6/QtQuickLayouts -isystem /usr/include/qt6/QtGui/6.7.1 -isystem /usr/include/qt6/QtGui/6.7.1/QtGui -isystem /usr/include/qt6/QtQuick/6.7.1 -isystem /usr/include/qt6/QtQuick/6.7.1/QtQuick -isystem /usr/include/qt6/QtQuickShapes -isystem /usr/include/qt6/QtQuickShapes/6.7.1 -isystem /usr/include/qt6/QtQuickShapes/6.7.1/QtQuickShapes -isystem /usr/include/qt6/QtQuickTemplates2 -isystem /usr/include/qt6/QtQuickWidgets -isystem /usr/include/qt6/QtQmlWorkerScript -isystem /usr/include/taglib -isystem /usr/include/qt6/QtDBus -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/qm-dsp -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/qm-dsp/include -isystem /usr/include/libupower-glib -isystem /usr/include/lilv-0 -isystem /usr/include/opus -isystem /usr/include/hidapi -isystem /usr/include/libusb-1.0 -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/xwax -isystem /usr/include/wavpack  -march=native -mtune=native -O2 -pipe -std=gnu++20 -fvisibility=hidden -fvisibility-inlines-hidden -pipe -Wall -Wextra -Woverloaded-virtual -Wfloat-conversion -Werror=return-type -Wformat=2 -Wformat-security -Wvla -Wundef -fmacro-prefix-map=/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999=. -pthread -fPIC -MD -MT CMakeFiles/mixxx-lib.dir/src/track/cueinfo.cpp.o -MF CMakeFiles/mixxx-lib.dir/src/track/cueinfo.cpp.o.d -o CMakeFiles/mixxx-lib.dir/src/track/cueinfo.cpp.o -c /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/track/cueinfo.cpp
FAILED: CMakeFiles/mixxx-lib.dir/src/track/cueinfo.cpp.o 
/usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++ -DMIXXX_BUILD_RELEASE -DMIXXX_USE_QML -DMIXXX_USE_QOPENGL -DPA_USE_ALSA -DPROTOBUF_USE_DLLS -DQT_CONCURRENT_LIB -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_LABSQMLMODELS_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_PRINTSUPPORT_LIB -DQT_QMLBUILTINS_LIB -DQT_QMLINTEGRATION_LIB -DQT_QMLMODELS_LIB -DQT_QMLWORKERSCRIPT_LIB -DQT_QML_LIB -DQT_QUICKCONTROLS2IMPL_LIB -DQT_QUICKCONTROLS2_LIB -DQT_QUICKLAYOUTS_LIB -DQT_QUICKSHAPES_LIB -DQT_QUICKTEMPLATES2_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_SHADERTOOLS_LIB -DQT_SQL_LIB -DQT_SVGWIDGETS_LIB -DQT_SVG_LIB -DQT_TABLET_SUPPORT -DQT_TESTCASE_BUILDDIR=\"/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999_build\" -DQT_TESTCASE_SOURCEDIR=\"/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999\" -DQT_TESTLIB_LIB -DQT_USE_QSTRINGBUILDER -DQT_WIDGETS_LIB -DQT_XML_LIB -DSFC_SUPPORTS_SET_COMPRESSION_LEVEL -D__BATTERY__ -D__BULK__ -D__FAAD__ -D__FFMPEG__ -D__HID__ -D__KEYFINDER__ -D__LILV__ -D__LINUX__ -D__MAD__ -D__MODPLUG__ -D__MP4V2__ -D__OPUS__ -D__PORTMIDI__ -D__QTKEYCHAIN__ -D__RUBBERBAND__ -D__SNDFILE__ -D__SQLITE3__ -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D__UNIX__ -D__VINYLCONTROL__ -D__WV__ -Dx86_64 -I/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999_build/mixxx-lib_autogen/include -I/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src -I/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999_build/src -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/fidlib -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/portaudio -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/rigtorp/SPSCQueue/include -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/replaygain -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/reverb -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/kaitai -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/mp3guessenc-0.27.4 -isystem /usr/include/qt6/QtConcurrent -isystem /usr/include/qt6 -isystem /usr/include/qt6/QtCore -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtOpenGL -isystem /usr/include/qt6/QtPrintSupport -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtQml -isystem /usr/include/qt6/QtQmlIntegration -isystem /usr/include/qt6/QtQmlBuiltins -isystem /usr/include/qt6/QtSql -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtTest -isystem /usr/include/qt6/QtXml -isystem /usr/include/qt6/QtShaderTools -isystem /usr/include/qt6/QtSvgWidgets -isystem /usr/include/qt6/QtCore5Compat -isystem /usr/include/qt6/QtQuick -isystem /usr/include/qt6/QtQmlModels -isystem /usr/include/qt6/QtLabsQmlModels -isystem /usr/include/qt6/QtQmlModels/6.7.1 -isystem /usr/include/qt6/QtQmlModels/6.7.1/QtQmlModels -isystem /usr/include/qt6/QtCore/6.7.1 -isystem /usr/include/qt6/QtCore/6.7.1/QtCore -isystem /usr/include/qt6/QtQml/6.7.1 -isystem /usr/include/qt6/QtQml/6.7.1/QtQml -isystem /usr/include/qt6/QtQmlBuiltins/6.7.1 -isystem /usr/include/qt6/QtQmlBuiltins/6.7.1/QtQmlBuiltins -isystem /usr/include/qt6/QtQuickControls2 -isystem /usr/include/qt6/QtQuickControls2Impl -isystem /usr/include/qt6/QtQuickLayouts -isystem /usr/include/qt6/QtGui/6.7.1 -isystem /usr/include/qt6/QtGui/6.7.1/QtGui -isystem /usr/include/qt6/QtQuick/6.7.1 -isystem /usr/include/qt6/QtQuick/6.7.1/QtQuick -isystem /usr/include/qt6/QtQuickShapes -isystem /usr/include/qt6/QtQuickShapes/6.7.1 -isystem /usr/include/qt6/QtQuickShapes/6.7.1/QtQuickShapes -isystem /usr/include/qt6/QtQuickTemplates2 -isystem /usr/include/qt6/QtQuickWidgets -isystem /usr/include/qt6/QtQmlWorkerScript -isystem /usr/include/taglib -isystem /usr/include/qt6/QtDBus -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/qm-dsp -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/qm-dsp/include -isystem /usr/include/libupower-glib -isystem /usr/include/lilv-0 -isystem /usr/include/opus -isystem /usr/include/hidapi -isystem /usr/include/libusb-1.0 -isystem /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/lib/xwax -isystem /usr/include/wavpack  -march=native -mtune=native -O2 -pipe -std=gnu++20 -fvisibility=hidden -fvisibility-inlines-hidden -pipe -Wall -Wextra -Woverloaded-virtual -Wfloat-conversion -Werror=return-type -Wformat=2 -Wformat-security -Wvla -Wundef -fmacro-prefix-map=/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999=. -pthread -fPIC -MD -MT CMakeFiles/mixxx-lib.dir/src/track/cueinfo.cpp.o -MF CMakeFiles/mixxx-lib.dir/src/track/cueinfo.cpp.o.d -o CMakeFiles/mixxx-lib.dir/src/track/cueinfo.cpp.o -c /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/track/cueinfo.cpp
/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/track/cueinfo.cpp: In function ‘QDebug mixxx::operator<<(QDebug, const CueInfo&)’:
/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/track/cueinfo.cpp:159:30: error: ambiguous overload for ‘operator<<’ (operand types are ‘QDebug’ and ‘std::optional<double>’)
  156 |     debug.nospace()
      |     ~~~~~~~~~~~~~~~           
  157 |             << "CueInfo["
      |             ~~~~~~~~~~~~~     
  158 |             << "type=" << cueInfo.getType()
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  159 |             << ", startPos=" << cueInfo.getStartPositionMillis()
      |             ~~~~~~~~~~~~~~~~ ^~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |             |                                                 |
      |             QDebug                                            std::optional<double>
In file included from /usr/include/qt6/QtCore/QtDebug:1,
                 from /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/audio/types.h:3,
                 from /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/audio/signalinfo.h:3,
                 from /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/track/cueinfo.h:5,
                 from /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/track/cueinfo.cpp:1:
/usr/include/qt6/QtCore/qdebug.h:385:34: note: candidate: ‘QDebugIfHasDebugStream<T> operator<<(QDebug, const std::optional<_Tp>&) [with T = double; QDebugIfHasDebugStream<T> = QDebug]’
  385 | inline QDebugIfHasDebugStream<T> operator<<(QDebug debug, const std::optional<T> &opt)
      |                                  ^~~~~~~~
In file included from /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/audio/types.h:8:
/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/util/optional.h:7:15: note: candidate: ‘QDebug operator<<(QDebug, std::optional<_Tp>) [with T = double]’
    7 | inline QDebug operator<<(QDebug dbg, std::optional<T> arg) {
      |               ^~~~~~~~


```

### Version

mixxx git main branch e666439609088508cf12aa48d76b870c64fa81e9

### OS

Gentoo linux

--- a/src/util/optional.h
+++ b/src/util/optional.h
@@ -1,6 +1,10 @@
 #pragma once
 
 #include <QtDebug>
+// support for QDebugging std::optional<T> has been added natively in Qt 6.7
+// this definition clashes with Qt's so disable it.
+#if QT_VERSION < QT_VERSION_CHECK(6, 7, 0)
+
 #include <optional>
 
 template<typename T>
@@ -11,3 +15,5 @@ inline QDebug operator<<(QDebug dbg, std::optional<T> arg) {
         return dbg << "nullopt";
     }
 }
+
+#endif

